internal/apicodec/codec_v2.go

```diff
@@ -953,3 +953,16 @@ func (c *codecV2) decodeLockInfos(locks []*kvrpcpb.LockInfo) ([]*kvrpcpb.LockInf
 	}
 	return locks, nil
 }
+
+func (c *codecV2) DecodeBucketKey(encodedKey []byte) ([]byte, error) {
+	key, err := c.memCodec.decodeKey(encodedKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if bytes.Compare(key, c.endKey) >= 0 || bytes.Compare(key, c.prefix) < 0 {
+		return []byte{}, nil
+	}
```

Comment:
should we return the corresponding prefix or endkey of the codec instead of return empty slice?

Reply:
This function aims for decoding the key, for example, if there is a region ranged in `[x0001, x0003)`, and the keyspace 002 tries to decode the bucket key "x0001test", then the key for keyspace 002 doesn't exist, we might decode them as "".

Reply:
got it